packages/cysync-core/src/constants/errors/types.ts

```diff
@@ -0,0 +1,30 @@
+import { ErrorIconType } from '@cypherock/cysync-ui';
+
+export const ErrorIconNameMap = {
+  device: 'device' as ErrorIconType,
+  default: 'default' as ErrorIconType,
+} as const;
+
+/**
+ * Only one of these actions should be specified
+ */
+export interface ErrorAction {
+  // Show only retry button
+  retry?: boolean;
+
+  // Show only report button
+  report?: boolean;
+
+  // Show both retry and report buttons
+  reportWithRetry?: boolean;
+
+  // Show both retry and report buttons, but only after retrying a certain number of times
+  reportAfterRetry?: {
+    maxRetries: number;
+  };
```

Comment:
Could we use an enum instead of multiple booleans? and `maxRetries` suggests user can't retry after this number it could be more along the lines of `retryThreshold` or something similar

Reply:
`maxRetries` means the user cannot retry after certain number of times.